test/bntest.c

```diff
@@ -2084,6 +2084,29 @@ static int test_smallprime()
     return st;
 }
 
+static int test_3_is_prime()
+{
+    int ret = 0;
+    BIGNUM *r = BN_new();
+
+    /* For a long time, small primes were not considered prime when
+     * do_trial_division was set. */
+    if (r == NULL ||
+        !BN_set_word(r, 3) ||
+        !BN_is_prime_fasttest_ex(r, 3 /* nchecks */, ctx,
+                                 0 /* do_trial_division */, NULL) ||
+        !BN_is_prime_fasttest_ex(r, 3 /* nchecks */, ctx,
+                                 1 /* do_trial_division */, NULL)) {
```

Comment:
`BN_is_prime_fasttest_ex` has a tri-state return, so this probably should be `!= 1`.